lib/templates/android_screen_base.tt

```diff
@@ -183,4 +183,25 @@ class AndroidScreenBase < Calabash::ABase
     # Clicking in the Done button
     touch "* id:'button1'"
   end
+  
+ 
+  
+  # @option [String] id The view's id (optional)
+  # @option [String] The complete view query
+  # @return [Array] return an array[String] associated with the views
+  def text_array_from(query, id = nil)
+    query_text = "* id:'#{id}'" unless id.nil?
+    query_text = query unless query.nil?
```

Comment:
You don't need to recheck it here. If your first parameter is an ID you can do this:

```
query = "* id:'#{id}'" if query.nil?
array_text = query(query, :text)
```

Since ID is a required parameter, you don't need another variable or check :)